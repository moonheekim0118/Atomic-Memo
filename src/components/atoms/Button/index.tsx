import React from 'react';
import styled from 'styled-components';

interface Props {
  children: React.ReactNode;
  color?: string;
  onClick?: (e: React.MouseEvent<HTMLSpanElement>) => void;
}

const Button = ({ children, color, onClick }: Props) => {
  return (
    <StyledButton onClick={onClick} color={color}>
      {children}
    </StyledButton>
  );
};

const StyledButton = styled.button<{ color: string }>`
  background-color: #fff;
  color: ${(props) => props.color};
  font-size: 1.2rem;
  font-weight: bold;
  padding: 0.3em;
  border: 2px solid #f4f4f4;
  border-radius: 10px;

  cursor: pointer;

  transition: all 0.5s ease;

  &:hover {
    background-color: ${(props) => props.color};
    border-color: ${(props) => props.color};
    color: #fff;
  }
  &:focus {
    outline: none;
  }
`;

export default Button;
