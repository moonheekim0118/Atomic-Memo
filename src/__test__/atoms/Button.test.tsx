import 'jsdom-global/register';
import React from 'react';
import Button from '../../components/atoms/Button';
import { mount } from 'enzyme';

describe('<Button/>', () => {
  let container = null;
  const mockFn = jest.fn();

  it('renders correctly', () => {
    container = mount(<Button onClick={mockFn}>test</Button>);
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should call onClick if Container is Clicked', () => {
    container.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});
