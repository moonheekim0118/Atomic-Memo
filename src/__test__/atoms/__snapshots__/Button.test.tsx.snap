// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> matches snpashot 1`] = `"<button class=\\"Button__StyledButton-hxb5gx-0 dXflyU\\">test</button>"`;
